### PR TITLE
Dodecahedron surface area and volume

### PR DESCRIPTION
**Pull request by shri30yans**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/dodecahedron.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    volume = ((15 + (7 * (5 ** (1 / 2)))) / 4) * (edge**3)
+    return volume
+
+
+if __name__ == "__main__":
+    edge = 5
+    print("Dodecahedron edge:", edge)
+    print("Surface area: ", dodecahedron_surface_area(edge=edge))
+    print("Volume: ", dodecahedron_volume(edge=edge))
```
Comment (ZeroDayOwl) [inline maths/dodecahedron.py:77]:
```suggestion
```

File: maths/dodecahedron.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    """
+
+    if edge <= 0 or not isinstance(edge, int):
+        raise ValueError("Length must be a positive.")
+
+    volume = ((15 + (7 * (5 ** (1 / 2)))) / 4) * (edge**3)
+    return volume
+
+
+if __name__ == "__main__":
```
Comment (ZeroDayOwl) [inline maths/dodecahedron.py:73]:
```suggestion
if __name__ == "__main__":
    import doctest
    doctest.testmod()
```

**Review by ZeroDayOwl — CHANGES_REQUESTED** [review]

File: maths/dodecahedron.py (hunk 2 above)
Comment (ZeroDayOwl) [reply maths/dodecahedron.py:73]: Incorporate those changes.

File: maths/dodecahedron.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    Traceback (most recent call last):
+      ...
+    ValueError: Length must be a positive.
+    """
+
+    if edge <= 0 or not isinstance(edge, int):
+        raise ValueError("Length must be a positive.")
+
+    volume = ((15 + (7 * (5 ** (1 / 2)))) / 4) * (edge**3)
+    return volume
```
Comment (cclauss) [inline maths/dodecahedron.py:70]:
```suggestion
    return ((15 + (7 * (5 ** (1 / 2)))) / 4) * (edge**3)
```

File: maths/dodecahedron.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    >>> dodecahedron_surface_area(-1)
+    Traceback (most recent call last):
+      ...
+    ValueError: Length must be a positive.
+    """
+
+    if edge <= 0 or not isinstance(edge, int):
+        raise ValueError("Length must be a positive.")
+    surface_area = 3 * ((25 + 10 * (5 ** (1 / 2))) ** (1 / 2)) * (edge**2)
+    return surface_area
```
Comment (cclauss) [inline maths/dodecahedron.py:38]:
```suggestion
    return 3 * ((25 + 10 * (5 ** (1 / 2))) ** (1 / 2)) * (edge**2)
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**